napari/_vispy/vispy_image_layer.py

```diff
@@ -93,6 +93,9 @@ def _on_data_change(self, event=None):
                 self.node.set_data(data)
             else:
                 self.node.set_data(data, clim=self.layer.contrast_limits)
+
+        # Call to update order of translation values with new dims:
+        self._reset_base()
```

Comment:
```suggestion
        self._on_scale_change()
        self._on_translate_change()
```
Looking at what is inside `_reset_base` I think you can just call these two methods and it will be fine

Reply:
Will be faster not to unnecessarily call the other methods

Reply:
Curious also @tlambert03 if we might want to put some sort of check inside `_on_scale_change ` to see if the scale has actually changed and only make the call if so, otherwise we might take a real performance hit for two unnecessary calls each time data is updated, or does vispy already do such a check under the hood?

Reply:
not immediately sure.  couldn't quickly find how `_on_scale_change` ultimately makes its way back to the vispy code...

Reply:
hmm, i think it is pretty easy to add something like `if np.all(self.scale == scale): return` in the scale setter https://github.com/napari/napari/blob/master/napari/_vispy/vispy_base_layer.py#L94, and similar for translate, then we don't run the risk. @vigji can you give this a go too?